From c2a6694fee3f3a7fc2e0cfccf415e8f0e8b52a21 Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Mon, 3 Dec 2012 17:15:21 +0800
Subject: [PATCH 377/635] ENGR00215174 fix allocate buffer problem.

add GRALLOC_USAGE_FORCE_CONTIGUOUS to allocate
continuous physical buffer.

Change-Id: If766a90d29e030ffcfdefdfee8dd421944455616
Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 mx6/libcamera/CameraUtil.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mx6/libcamera/CameraUtil.h b/mx6/libcamera/CameraUtil.h
index ca3b3a7..2a50ac2 100644
--- a/mx6/libcamera/CameraUtil.h
+++ b/mx6/libcamera/CameraUtil.h
@@ -81,7 +81,8 @@ using namespace android;
 #define CAMERA_GRALLOC_USAGE GRALLOC_USAGE_HW_TEXTURE | \
     GRALLOC_USAGE_HW_RENDER |                           \
     GRALLOC_USAGE_SW_READ_RARELY |                      \
-    GRALLOC_USAGE_SW_WRITE_NEVER
+    GRALLOC_USAGE_SW_WRITE_NEVER |                      \
+    GRALLOC_USAGE_FORCE_CONTIGUOUS
 
 #define CAMERA_MAX(x, y) (x) > (y) ? (x) : (y)
 
-- 
1.8.0

